### PR TITLE
Add join table

### PR DESCRIPTION
**Pull request by abiola-arobieke**
In this milestone:

Added new tables: visits, specializations, and vets tables to the schema.sql file.
Inserted data into the new tables.
Added queries to query the tables.

Attached below are screenshots of the query results;

Screenshot 2023-06-23 at 19 49 24

Screenshot 2023-06-23 at 20 17 09

Screenshot 2023-06-23 at 20 41 02

Screenshot 2023-06-23 at 20 57 09

Screenshot 2023-06-23 at 21 21 05

Screenshot 2023-06-23 at 21 59 57

Screenshot 2023-06-24 at 10 03 00

Screenshot 2023-06-24 at 11 48 26

Screenshot 2023-06-29 at 16 40 55

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -35,4 +35,43 @@ ALTER TABLE animals ADD CONSTRAINT fk_species_id FOREIGN KEY (species_id) REFERE
...
+    date_of_graduation DATE NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE vets (id INT GENERATED ALWAYS AS IDENTITY,
+    name VARCHAR(250) NOT NULL,
+    age INT NOT NULL,
+    date_of_graduation DATE NULL,
+    PRIMARY KEY(id)
+);
```
Comment (AnselemOdims) [inline schema.sql]: - [x] There are two commands to create the vets table. Please kindly remove one

File: queries.sql [hunk 2]
```diff
@@ -51,3 +51,46 @@ SELECT species.name, COUNT(species.name) AS number_of_animals FROM animals JOIN
...
+    specie_id INT REFERENCES species(id),
+    PRIMARY KEY (vet_id, specie_id)
+);
+
+CREATE TABLE visits (
+    animal_id INT REFERENCES animals(id),
+    vet_id INT REFERENCES vets(id),
+    visit_date DATE,
+    PRIMARY KEY (animal_id, vet_id, visit_date)
+);
```
Comment (AnselemOdims) [inline queries.sql]: - [x] I noticed that these commands to create tables have already been included in the schema.sql file. Please kindly remove it from this file.

File: queries.sql [hunk 3]
```diff
@@ -51,3 +51,46 @@ SELECT species.name, COUNT(species.name) AS number_of_animals FROM animals JOIN
...
+
+Who was the last animal seen by William Tatcher?
+How many different animals did Stephanie Mendez see?
+List all vets and their specialties, including vets with no specialties.
+List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+What animal has the most visits to vets?
+Who was Maisy Smith's first visit?
+Details for most recent visit: animal information, vet information, and date of visit.
+How many visits were with a vet that did not specialize in that animal's species?
+What specialty should Maisy Smith consider getting? Look for the species she gets the most.
```
Comment (AnselemOdims) [inline queries.sql]: - [x] Good job outlining the query commands for this milestone, but please kindly include the actual SQL query commands as required.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @abiola-arobieke , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: schema.sql (hunk 1 above)
Comment (rloterh) [reply schema.sql]: well resolved! ✔️

File: queries.sql (hunk 2 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

File: queries.sql (hunk 3 above)
Comment (rloterh) [reply queries.sql]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Abiola,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on implementing recommended changes!
🎲 ReadMe looks great!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
One screenshot of the queries was omitted. Please let take of this and be more observant on subsequent occasion! 😉 

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**